python/cuml/random_projection/rproj.pyx

```diff
@@ -0,0 +1,392 @@
+        cdef uintptr_t output_ptr = self._get_ctype_ptr(X_new)
+
+        if self.params.n_features != n_features:
+            raise ValueError("n_features must be same as on fitting: %d" %
+                         self.params.n_features)
+
+        if self.gdf_datatype.type == np.float32:
+            RPROJtransform[float](<float*> input_ptr,
+                        self.rand_matS,
+                        <float*> output_ptr,
+                        &self.params)
+        else:
+            RPROJtransform[double](<double*> input_ptr,
+                        self.rand_matD,
+                        <double*> output_ptr,
+                        &self.params)
+
+        if (isinstance(X, cudf.DataFrame)):
+            del(X_m)
+
+        return X_new
+
+
+class GaussianRandomProjection(BaseRandomProjection):
+    """
+    Gaussian Random Projection method derivated from BaseRandomProjection class.
+
+    Random projection is a dimensionality reduction technique. Random projection methods
+    are powerful methods known for their simplicity, computational efficiency and restricted model size.
+    This algorithm also has the advantage to preserve distances well between any two samples
+    and is thus suitable for methods having this requirement.
+
+    The components of the random matrix are drawn from N(0, 1 / n_components).
+
+    Example
+    ---------
+
+    .. code-block:: python
+        from cuml.random_projection import GaussianRandomProjection
+        from sklearn.datasets.samples_generator import make_blobs
+        from sklearn.svm import SVC
+
+        # dataset generation
+        data, target = make_blobs(n_samples=800, centers=400, n_features=3000, random_state=0)
+
+        # model fitting
+        model = GaussianRandomProjection(n_components=5, random_state=42).fit(data)
+
+        # dataset transformation
+        transformed_data = model.transform(data)
+
+        # classifier training
+        classifier = SVC(gamma=0.001).fit(transformed_data, target)
+
+        # classifier scoring
+        score = classifier.score(transformed_data, target)
+
+        # measure information preservation
+        print("Score: {}".format(score))
+
+    Output:
+
+    .. code-block:: python
+        Score: 1.0
+
+    Parameters
+    ----------
+
+    n_components : int (default = 'auto')
+        Dimensionality of the target projection space. If set to 'auto',
+        the parameter is deducted thanks to Johnson–Lindenstrauss lemma.
+        The automatic deduction make use of the number of samples and
+        the eps parameter.
+
+        The Johnson–Lindenstrauss lemma can produce very conservative
+        n_components parameter as it makes no assumption on dataset structure.
+
+    eps : float (default = 0.1)
+        Error tolerance during projection. Used by Johnson–Lindenstrauss
+        automatic deduction when n_components is set to 'auto'.
+
+    random_state : int (default = 42)
```

Comment:
As much as I hate to suggest this change since I love your default :), shouldn't the default be `None`?

Reply:
^^ Yes, will change it